### PR TITLE
Run nightly Clippy over the Ruff repo

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

This is the result of running `cargo +nightly clippy --workspace --all-targets --all-features -- -D warnings` and fixing all violations. Just wanted to see if there were any interesting new checks on nightly :eyes:

### CODE REVIEW

File: crates/ruff/src/rules/isort/order.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ pub(crate) fn order_imports<'a>(
             )
             .chain(
                 // Include all star imports.
-                block.import_from_star.into_iter(),
```
Comment (charliermarsh) [inline crates/ruff/src/rules/isort/order.rs:50]: (Calling `.into_inter()` on a method that accepts `IntoIter`.)

**Review by zanieb — APPROVED** [review]
Seems reasonable. It'd be cool to automate this but if it can't autofix most of the violations it seems tricky.

**Comment by charliermarsh** [comment]
Eventually these rules will make it out of nightly and into subsequent Rust releases, at which point we'll be able to incorporate them into our existing Clippy checks for free :)

**Comment by MichaReiser** [comment]
Uhm what. No approve button on github mobile 🧐 *Approve*

**Comment by dhruvmanila** [comment]
> Uhm what. No approve button on github mobile 🧐 _Approve_

It's hidden inside the file section > "Review Changes" button ;)

**Review by MichaReiser — COMMENTED** [review]
Damn, I never hit Submit on these comments

File: crates/ruff_python_formatter/src/generated.rs [hunk 2]
```diff
@@ -35,7 +35,7 @@ impl<'ast> IntoFormat<PyFormatContext<'ast>> for ast::ModModule {
         PyFormatContext<'ast>,
     >;
     fn into_format(self) -> Self::Format {
-        FormatOwnedWithRule::new(self, crate::module::mod_module::FormatModModule::default())
+        FormatOwnedWithRule::new(self, crate::module::mod_module::FormatModModule)
```
[the remark this answers is not in the document]
Comment (MichaReiser) [reply crates/ruff_python_formatter/src/generated.rs:38]: I'm gonna revert the generated changes. See https://github.com/astral-sh/ruff/pull/5804 for why

**Pull request merged**